modules/auxiliary/gather/ldap_esc_vulnerable_cert_finder.rb

```diff
@@ -520,6 +712,41 @@ def find_esc15_vuln_cert_templates
     query_ldap_server_certificates(esc_raw_filter, 'ESC15', notes: notes)
   end
 
+  def find_esc16_vuln_cert_templates
+    esc16_raw_filter = '(&'\
+      '(objectclass=pkicertificatetemplate)'\
+      '(!(mspki-enrollment-flag:1.2.840.113556.1.4.804:=2))'\
+      '(|(mspki-ra-signature=0)(!(mspki-ra-signature=*)))'\
+      '(pkiextendedkeyusage=*)'\
+    ')'
+
+    esc_entries = query_ldap_server(esc16_raw_filter, CERTIFICATE_ATTRIBUTES, base_prefix: CERTIFICATE_TEMPLATES_BASE)
+    return if esc_entries.empty?
+
+    registry_object = @ldap_objects.find { |obj| obj[:type] == :registry_values }
+
+    strong_binding = registry_object[:values][:strong_certificate_binding_enforcement]
+    if strong_binding && (strong_binding.to_i == 0 || strong_binding.to_i == 1)
+      # Scenario 1 -  StrongCertificateBindingEnforcement = 1 or 0 then it's same same as ESC9 - mark them all as vulnerable
+      esc_entries.each do |entry|
+        certificate_symbol = entry[:cn][0].to_sym
+        @certificate_details[certificate_symbol][:techniques] << 'ESC16'
+        @certificate_details[certificate_symbol][:notes] << "ESC16: Template is vulnerable due StrongCertificateBindingEnforcement = #{registry_object[:values][:strong_certificate_binding_enforcement]} and the CA's disabled policy extension list includes: 1.3.6.1.4.1.311.25.2."
+      end
+    else
+      # Scenario 2 - StrongCertificateBindingEnforcement = 2 (or nil) but if EditFlags in the active policy module has the bit 0x00040000 set then ESC6 is essentially re-enabled and we mark them all as vulnerable
+      edit_flags = registry_object[:values][:edit_flags]
+      if edit_flags.to_i & 0x00040000 != 0
+        esc_entries.each do |entry|
+          certificate_symbol = entry[:cn][0].to_sym
+          @certificate_details[certificate_symbol][:techniques] << 'ESC16'
+          @certificate_details[certificate_symbol][:notes] << 'ESC16: Template is vulnerable due to the active policy EditFlags having the bit: 0x00040000 set (which is essentially ESC6) combined with the CA\'s disabled policy extension list including: 1.3.6.1.4.1.311.25.2.'
```

Comment:
Is there a contant we can add to and use from MsCrtd for `0x00040000`? That'd also make the description more readable.